services/server/src/server/apiv1/verification/solc-json/stateless/solc-json.stateless.handlers.ts

```diff
@@ -38,59 +35,44 @@ export async function verifySolcJsonEndpoint(req: Request, res: Response) {
   const chain = req.body.chain;
   const address = req.body.address;
 
+  // Get metadata and sources from the Solidity JSON input
   const metadataAndSourcesPathBuffers =
     await services.verification.getAllMetadataAndSourcesFromSolcJson(
       solc,
       solcJson,
       compilerVersion,
     );
 
-  const checkedContracts = await checkFilesWithMetadata(
-    solc,
-    vyper,
+  // Create metadata contracts from the files
+  const metadataContracts = await createMetadataContractsFromFiles(
     metadataAndSourcesPathBuffers,
   );
```

Comment:
Why are we generating metadata files here? Can't we directly use the solcJson we have?